docker-compose.prod.yaml

```
@@ -0,0 +1,89 @@
+services:
+  zero:
+    build:
+      context: .
+      dockerfile: docker/app/Dockerfile
+      args:
+        NEXT_PUBLIC_APP_URL: ${NEXT_PUBLIC_APP_URL:-http://localhost:3000}
+        DATABASE_URL: postgresql://${POSTGRES_USER:-postgres}:${POSTGRES_PASSWORD:-postgres}@db:5432/${POSTGRES_DB:-zerodotemail}
+        BETTER_AUTH_SECRET: ${BETTER_AUTH_SECRET:-my-better-auth-secret}
+        BETTER_AUTH_URL: ${BETTER_AUTH_URL:-http://localhost:3000}
+        BETTER_AUTH_TRUSTED_ORIGINS: ${BETTER_AUTH_TRUSTED_ORIGINS:-http://localhost:3000}
+        GOOGLE_CLIENT_ID: ${GOOGLE_CLIENT_ID}
+        GOOGLE_CLIENT_SECRET: ${GOOGLE_CLIENT_SECRET}
+        REDIS_URL: http://upstash-proxy:80
+        REDIS_TOKEN: ${REDIS_TOKEN:-upstash-local-token}
+        RESEND_API_KEY: ${RESEND_API_KEY}
+        OPENAI_API_KEY: ${OPENAI_API_KEY}
+        AI_SYSTEM_PROMPT: ${AI_SYSTEM_PROMPT}
+        GROQ_API_KEY: ${GROQ_API_KEY}
+        GOOGLE_GENERATIVE_AI_API_KEY: ${GOOGLE_GENERATIVE_AI_API_KEY}
+    depends_on:
+      - db
+      - migrations
+      - valkey
+      - upstash-proxy
```

```suggestion
    depends_on:
      db:
        condition: service_healthy
      valkey:
        condition: service_healthy
      upstash-proxy:
        condition: service_healthy
```

1. Not sure if having `- migrations` as a dependency for the Zero container will cause the container to stuck on starting or unhealthy state.

2. I think it's good to have the condition `service_healthy` so the Zero container won't start if some other dependency fails to start for whatever reason